File: python_files/InvVariance.py
import networkx as nx
import matplotlib.pyplot as plt
import time

def mean_degree(G):
    if G.number_of_nodes() == 1:
        return 0
    count = 0
    for node in G:
        count += G.degree(node)
    return count/G.number_of_nodes()

def var_degree(G):
    if G.number_of_nodes() == 1:
        return 0
    mean = mean_degree(G)
    count = 0
    for node in G:
        count += (G.degree(node) - mean)**2
    return count/(G.number_of_nodes()-1)
    
    

File: rust/src/inv_variance.rs
use crate::graph::UGraph;

pub fn mean_degree(g:UGraph)->f32{
    let n = g.number_of_nodes();
    if n < 2{
        return 0.0;
    } 
    let mut count = 0;
    for node in &g.nodes{
        count += &g.degree(*node);}
    (count as f32) / (n as f32)
}

pub fn var_degree(g:UGraph)->f32{
    let n = g.number_of_nodes();
    if n < 2 {
        return 0.0;
    }
    let mean = mean_degree(g.clone());
    let mut count = 0.0;
    for node in &g.nodes{
        count += ((g.degree(*node) as f32) -mean).powf(2.0);
    }
    (count as f32) / ((n - 1) as f32)
}

